Sources/Components/Components/Empty View/WKEmptyView.swift

```diff
@@ -0,0 +1,91 @@
+import SwiftUI
+
+public protocol WKEmptyViewDelegate: AnyObject {
+    func emptyViewDidTapSearch()
+    func emptyViewDidTapFilters()
+}
+
+public struct WKEmptyView: View {
+
+    @ObservedObject var appEnvironment = WKAppEnvironment.current
+    @ObservedObject var viewModel: WKEmptyViewModel
+    var delegate: WKEmptyViewDelegate?
+    var type: WKEmptyViewStateType
+
+    public var body: some View {
+        GeometryReader { geometry in
+
+            ZStack {
+                Color(appEnvironment.theme.paperBackground)
+                    .ignoresSafeArea()
+                ScrollView {
+                    VStack {
+                        Spacer()
+                        Image(uiImage: viewModel.image)
+                            .frame(width: 132, height: 118)
+                        Text(viewModel.localizedStrings.title)
+                            .font(Font(WKFont.for(.boldBody)))
+                            .foregroundColor(Color(appEnvironment.theme.text))
+                            .padding([.top], 12)
+                            .padding([.bottom], 8)
+                            .multilineTextAlignment(.center)
+                        if type == .filter {
+                            WKEmptyViewFilterView(delegate: delegate, viewModel: viewModel)
+                        } else {
+                            Text(viewModel.localizedStrings.subtitle)
+                                .font(Font(WKFont.for(.footnote)))
+                                .foregroundColor(Color(appEnvironment.theme.secondaryText))
+                                .multilineTextAlignment(.center)
+                        }
+                        if type == .noItems {
+                            WKSecondaryButton(title: viewModel.localizedStrings.buttonTitle, action: delegate?.emptyViewDidTapSearch)
+                                .padding([.leading, .trailing], 32)
+                        }
+                        Spacer()
+                    }
+                    .frame(minHeight: geometry.size.height)
+                    .padding([.leading, .trailing], 32)
+                }
+            }
+        }
+    }
+
+}
+
+struct WKEmptyViewFilterView: View {
+
+    @ObservedObject var appEnvironment = WKAppEnvironment.current
+    var delegate: WKEmptyViewDelegate?
+    @ObservedObject var viewModel: WKEmptyViewModel
+
+    var body: some View {
+
+        if #available(iOS 15, *) { //TODO: remove check after updating deployment target
+            var attributedString: AttributedString {
+                let str1 = viewModel.localizedStrings.filterSubtitleModify + " "
```

Comment:
See my comment [here](https://github.com/wikimedia/wikipedia-ios/pull/4616#discussion_r1320154047) about splicing issues. I'd rather the AttributedString come entirely from the app, if possible. But I'm not sure if we can get the range coloring to work.

Reply:
Thanks for helping me crack this one!